### ISSUE TITLE
pkey causes timeout errors and forces the entire scan to fail

### ISSUE THREAD

**Issue by sodejm** [jonluca/Anubis#21]
# Please follow the general troubleshooting steps first:

- [ Yes] Check you are using Python 3.6
- [ Yes] Make sure you are using pip3 (or any instance of pip associated with Python 3.x)

### Bug reports:

Make sure you include the following:

* Anubis Version - 0.6.4
* Python Version - 3.6.3
* Full stack trace if crash or installation error
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/connection.py", line 142, in _new_conn
    (self.host, self.port), self.timeout, **extra_kw)
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/util/connection.py", line 91, in create_connection
    raise err
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/util/connection.py", line 81, in create_connection
    sock.connect(sa)
TimeoutError: [Errno 60] Operation timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/connectionpool.py", line 578, in urlopen
    chunked=chunked)
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/connectionpool.py", line 351, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/connectionpool.py", line 814, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/connection.py", line 254, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/connection.py", line 151, in _new_conn
    self, "Failed to establish a new connection: %s" % e)
requests.packages.urllib3.exceptions.NewConnectionError: <requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x10f246828>: Failed to establish a new connection: [Errno 60] Operation timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/requests/adapters.py", line 403, in send
    timeout=timeout
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/connectionpool.py", line 623, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/util/retry.py", line 281, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
requests.packages.urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='www.pkey.in', port=443): Max retries exceeded with url: /tools-i/search-subdomains (Caused by NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x10f246828>: Failed to establish a new connection: [Errno 60] Operation timed out',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/anubis", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/anubis/cli.py", line 115, in main
    command.run()
  File "/usr/local/lib/python3.6/site-packages/anubis/commands/target.py", line 76, in run
    Thread(target=search_pkey(self, self.options["TARGET"][i])),
  File "/usr/local/lib/python3.6/site-packages/anubis/scanners/pkey.py", line 18, in search_pkey
    headers=headers, data=data, verify=False)
  File "/usr/local/lib/python3.6/site-packages/requests/api.py", line 111, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/requests/api.py", line 57, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/requests/sessions.py", line 475, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.6/site-packages/requests/sessions.py", line 585, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/requests/adapters.py", line 467, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='www.pkey.in', port=443): Max retries exceeded with url: /tools-i/search-subdomains (Caused by NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x10f246828>: Failed to establish a new connection: [Errno 60] Operation timed out',))
```
* Machine details (OS, build number, distro, etc.)
```
macOS 10.13.2
```

It looks like https://pkey.in times out which causes urllib3 to get a timeout exception. Once this exception occurs the scan times out and quits. I suggest removing pkey or creating an easy way to disable that specific searcher through a command line flag. Additionally there may need to be additional error handling to recover from these issues.

Note: pkey.in is also timing out in my browser.

**Comment by jonluca** [comment]
I hope Anubis didn't accidentally take down pkey. 

Anyways, should be fixed in 0.6.5. Thanks for the report!

**Comment by sodejm** [comment]
NP, thanks for the awesome tool and quick fix.
